### PR TITLE
Demo slack bot initial commit

### PR DESCRIPTION
**Pull request by ianplunkett**
Medplum Demo Bot that sends a Slack message to the “#general” channel when triggered.

### CODE REVIEW

File: examples/medplum-demo-bots/src/slack-bots/README.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+2. Navigate to the Bot creation page.
+3. Create a new bot or select an existing one to edit.
+4. In the bot's code editor, paste the contents of the `demo-slack-bot.ts` file.
+5. Save the bot code.
+
+For further reference on creating a bot see [Medplum Demo Bots](https://github.com/medplum/medplum/tree/main/examples/medplum-demo-bots)
+
+### 3. Configure Bot Secrets
+
+1. In the Medplum bot configuration page, find the "Secrets" section.
```
Comment (rahul1) [inline examples/medplum-demo-bots/src/slack-bots/README.md:36]: This is a bit misleading. The secretes are not really in the "Bot" configuration, they're in the Project configuration. That is, they are set once per project and shared across Bots. 

See this doc for more details: https://www.medplum.com/docs/access/projects#settings

File: examples/medplum-demo-bots/src/slack-bots/README.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+  - Check that the specified Slack channel exists and the bot has access to it.
+- For more detailed error information, review the bot's logs in Medplum.
+
+## Further Customization
+
+You can modify the `slackMessage` in the `handler` function to customize the content of the Slack notifications based on your specific needs.
+
+## Support
+
+For issues related to Medplum, please refer to the [Medplum documentation](https://www.medplum.com/docs) or contact Medplum support.
```
Comment (rahul1) [inline examples/medplum-demo-bots/src/slack-bots/README.md:75]:
```suggestion
For issues related to Medplum, please refer to the [Medplum documentation](https://www.medplum.com/docs) or contact Medplum support in [Discord](https://discord.gg/medplum)
```

**Review by rahul1 — APPROVED** [review]
Some minor readme comments

File: examples/medplum-demo-bots/src/slack-bots/README.md (hunk 1 above)
Comment (ianplunkett) [reply examples/medplum-demo-bots/src/slack-bots/README.md]: @rahul1 I've updated this part to refer to project level secrets but still point to https://www.medplum.com/docs/bots/bot-secrets for further reference.

**Pull request merged**